### PR TITLE
feat: add two new verify image policies

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

## What type of PR is this

/kind feature

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/kyverno/kyverno/blob/main/CONTRIBUTING.md).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] My PR contains new or altered behavior to Kyverno and
  
  - [x] I have read the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) and followed the process including adding proof manifests to this PR.

### CODE REVIEW

File: other/verify_image gcpkms.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: verify-image-gcpkms
+  annotations:
+    policies.kyverno.io/title: Verify Image
```
Comment (chipzoller) [inline other/verify_image gcpkms.yaml:6]: The title annotation is what the website filter will scrape, so these names should be unique.

File: other/verify_image gcpkms.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    policies.kyverno.io/category: Sample
+    policies.kyverno.io/severity: medium
+    policies.kyverno.io/subject: Pod
+    policies.kyverno.io/minversion: 1.4.2
```
Comment (chipzoller) [inline other/verify_image gcpkms.yaml:10]: This sets the minimum compatible version of Kyverno. Use of this new gcpkms key was added later, was it not?

File: other/verify_image gcpkms.yaml (hunk 1 above)
Comment (chipzoller) [reply other/verify_image gcpkms.yaml:6]: And there's already that name [here](https://kyverno.io/policies/other/verify_image/).

File: other/verify_image_k8s.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  background: false
+  rules:
+    - name: verify-image
+      match:
+        resources:
+          kinds:
+            - Pod
+      verifyImages:
+      - image: "ghcr.io/kyverno/test-verify-image:*"
+        key: default/cosign-keys
```
Comment (chipzoller) [inline other/verify_image_k8s.yaml:29]: Where is this? How is it set? Did this exist in 1.4.2? Does it require prerequisites to use this policy?

**Review by chipzoller — CHANGES_REQUESTED** [review]

**Comment by chipzoller** [comment]
@developer-guy Will you please address comments?

**Comment by developer-guy** [comment]
> @developer-guy Will you please address comments?

Of course, I'll do it ASAP 🙋🏻‍♂️

File: other/verify_image_k8s.yaml (hunk 3 above)
Comment (developer-guy) [reply other/verify_image_k8s.yaml:29]: yes, users might generate key pairs first by specifying k8s:// schema while using the `cosign` generate-key-pair command.

@JimBugwadia can you help me a bit, is [PR](https://github.com/kyverno/kyverno/pull/2607/) released? I couldn't see this PR in the changelogs of the releases in Kyverno?

File: other/verify_image gcpkms.yaml (hunk 1 above)
Comment (developer-guy) [reply other/verify_image gcpkms.yaml:6]: Thank you so much, fixed.

File: other/verify_image gcpkms.yaml (hunk 2 above)
Comment (developer-guy) [reply other/verify_image gcpkms.yaml:10]: As I mentioned below, with this [PR](https://github.com/kyverno/kyverno/pull/2607/), we added that support, but it seems not been released yet.

File: other/verify_image_k8s.yaml (hunk 3 above)
Comment (chipzoller) [reply other/verify_image_k8s.yaml:29]: This will be a 1.6.0 feature

**Comment by chipzoller** [comment]
These sample policies will require Kyverno 1.6.0 and will be held until then.

**Comment by chipzoller** [comment]
Any more modification/testing needed here or are they ready to go?

**Comment by chipzoller** [comment]
@JimBugwadia do these look good to you now?

File: other/verify_image_k8s.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      ghcr.io/kyverno/test-verify-image to ensure it has been signed by verifying
+      its signature against the provided public key. This policy serves as an illustration for
+      how to configure a similar rule and will require replacing with your image(s) and keys.
+spec:
+  validationFailureAction: enforce
+  background: false
+  rules:
+    - name: verify-image
+      match:
+        resources:
```
Comment (chipzoller) [inline other/verify_image_k8s.yaml:24]: Need to insert `any` here as `match` blocks have deprecated not using either `any`|`all`.

File: other/verify_image gcpkms.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      ghcr.io/kyverno/test-verify-image to ensure it has been signed by verifying
+      its signature against the provided public key. This policy serves as an illustration for
+      how to configure a similar rule and will require replacing with your image(s) and keys.
+spec:
+  validationFailureAction: enforce
+  background: false
+  rules:
+    - name: verify-image
+      match:
+        resources:
```
Comment (chipzoller) [inline other/verify_image gcpkms.yaml:24]: Need to insert `any` here as `match` blocks have deprecated not using either `any`|`all`.

File: other/verify_image gcpkms.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    policies.kyverno.io/minversion: 1.4.2
+    policies.kyverno.io/description: >-
+      Using the Cosign project, OCI images may be signed to ensure supply chain
+      security is maintained. Those signatures can be verified before pulling into
+      a cluster. This policy checks the signature of an image repo called
+      ghcr.io/kyverno/test-verify-image to ensure it has been signed by verifying
+      its signature against the provided public key. This policy serves as an illustration for
+      how to configure a similar rule and will require replacing with your image(s) and keys.
+spec:
+  validationFailureAction: enforce
```
Comment (chipzoller) [inline other/verify_image gcpkms.yaml:19]: Please set validationFailureAction to `audit` to be consistent with most policies here.

File: other/verify_image_k8s.yaml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    policies.kyverno.io/minversion: 1.4.2
+    policies.kyverno.io/description: >-
+      Using the Cosign project, OCI images may be signed to ensure supply chain
+      security is maintained. Those signatures can be verified before pulling into
+      a cluster. This policy checks the signature of an image repo called
+      ghcr.io/kyverno/test-verify-image to ensure it has been signed by verifying
+      its signature against the provided public key. This policy serves as an illustration for
+      how to configure a similar rule and will require replacing with your image(s) and keys.
+spec:
+  validationFailureAction: enforce
```
Comment (chipzoller) [inline other/verify_image_k8s.yaml:19]: Please set validationFailureAction to `audit` to be consistent with most policies here.

**Review by chipzoller — CHANGES_REQUESTED** [review]
Couple minor tweaks, if you please.

File: other/verify_image_k8s.yaml (hunk 4 above)
Comment (developer-guy) [reply other/verify_image_k8s.yaml:24]: thank you 🙋🏻‍♂️

File: other/verify_image gcpkms.yaml (hunk 5 above)
Comment (developer-guy) [reply other/verify_image gcpkms.yaml]: thank you 🙋🏻‍♂️

File: other/verify_image gcpkms.yaml (hunk 6 above)
Comment (developer-guy) [reply other/verify_image gcpkms.yaml]: thank you 🙋🏻‍♂️

File: other/verify_image_k8s.yaml (hunk 7 above)
Comment (developer-guy) [reply other/verify_image_k8s.yaml]: thank you 🙋🏻‍♂️

**Review by chipzoller — APPROVED** [review]
Thanks!

**Pull request merged**